### PR TITLE
Add support for node/input/output tooltips

### PR DESCRIPTION
**Pull request by pythongosssss**
Vue implementation of node tooltips

### CODE REVIEW

File: src/components/graph/NodeTooltip.vue [hunk 1]
```diff
@@ -0,0 +1,168 @@
+<template>
+  <div
+    v-if="tooltipText"
+    ref="tooltipRef"
+    class="node-tooltip"
+    :style="{ left, top }"
+  >
+    {{ tooltipText }}
+  </div>
+</template>
```
Comment (huchenlei) [inline src/components/graph/NodeTooltip.vue:10]: PrimeVue has [tooltip directive](https://primevue.org/tooltip/) and [popover component](https://primevue.org/tooltip/) available. Maybe we can try take advantage of existing component next time.

File: src/components/graph/NodeTooltip.vue [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+
+<script setup lang="ts">
+import { nextTick, ref, onBeforeUnmount, watch } from 'vue'
+import { LiteGraph } from '@comfyorg/litegraph'
+import { app as comfyApp } from '@/scripts/app'
+import { useNodeDefStore } from '@/stores/nodeDefStore'
+import { useSettingStore } from '@/stores/settingStore'
+
+let idleTimeout: NodeJS.Timeout
```
Comment (huchenlei) [inline src/components/graph/NodeTooltip.vue]: I think this should just be `number` as we are in the browser environment.

File: src/types/apiTypes.ts [hunk 3]
```diff
@@ -272,6 +272,7 @@ const zComfyNodeDef = z.object({
   output: zComfyOutputTypesSpec,
   output_is_list: z.array(z.boolean()),
   output_name: z.array(z.string()),
+  output_tooltips: z.optional(z.array(z.string())),
```
Comment (huchenlei) [inline src/types/apiTypes.ts]: Can be `z.array(z.string()).optional()` to match rest of schema definition.

File: src/components/graph/NodeTooltip.vue (hunk 1 above)
Comment (pythongosssss) [reply src/components/graph/NodeTooltip.vue:10]: I looked at those and neither seem particularly useful as the tooltip doesnt seem to have a documented API for specifing a position & showing it, and the popover wraps an element which would mean I need to position the tooltip in the correct position anyway. If im missing something there (didnt want to dig through the source), lmk and I'm happy to swap it.

File: src/components/graph/NodeTooltip.vue (hunk 2 above)
Comment (pythongosssss) [reply src/components/graph/NodeTooltip.vue]: image
I would then need to cast it

Comment (pythongosssss) [reply src/components/graph/NodeTooltip.vue]: `ReturnType<typeof setTimeout>` is probably the other option, but that will still be NodeJS.Timeout

File: src/components/graph/NodeTooltip.vue (hunk 1 above)
Comment (huchenlei) [reply src/components/graph/NodeTooltip.vue:10]: Popover's show method accept an event and uses the event target's location to position the overlay. We can simulate an event to use it. Tooltip directive might be more tricky to utilize as it requires HTML DOM element to anchor to, and creating so many invisible DOM elements overlay might cause peef issue.

Since you already have a functioning impl ready, I don't think we need to bend ourself to any of the solution above-mentioned. Just for next time, we should access what is already available. PrimeVue is pretty flexible. If passtheough is not enough, you can even inherit the component to get more freedom like I did for Autocomplete and Tree under PrimeVue override for.

File: src/components/graph/NodeTooltip.vue (hunk 2 above)
Comment (huchenlei) [reply src/components/graph/NodeTooltip.vue]: You can explicitly call `window.setTimeout` here.

**Review by huchenlei — APPROVED** [review]
LGTM

**Comment by huchenlei** [comment]
Ref: backend pr https://github.com/comfyanonymous/ComfyUI/pull/3842

**Comment by Amorano** [comment]
This looks solid. Thank you for moving the tooltip meta field(s). Also thank you for a node tooltip.

I feel the tooltip timeout might be better suited to being adjustable. 500 feels too short (for me) and I like to use 750+ -- seems like an easy setting expose.

thank you for the update and changes!

**Comment by Extraltodeus** [comment]
I'm currently testing and I'm not seeing anything appear on any node. Even after restarting the browser (currently firefox) and making sure that the node I was testing was a newly created one.

```
class EmptyRGBImage:
    @classmethod
    def INPUT_TYPES(s):
        return {"required": { "width": ("INT", {"default": 1024, "min": 1, "max": 16384, "step": 1, "tooltip":"Test"}),
                              "height": ("INT", {"default": 1024, "min": 1, "max": 16384, "step": 1, "tooltip":"Test plop"}),
                              "batch_size": ("INT", {"default": 1, "min": 1, "max": 4096}),
                              "r": ("INT", {"default": 0, "min": 0, "max": 255, "step": 1, "tooltip":"Surely it must be about the red color."}),
                              "g": ("INT", {"default": 0, "min": 0, "max": 255, "step": 1, "tooltip":"A color related to something needing to be touched."}),
                              "b": ("INT", {"default": 0, "min": 0, "max": 255, "step": 1, "tooltip":"The sky, the sea and my heart when you're not here."}),
                              }}
    RETURN_TYPES = ("IMAGE",)
    FUNCTION = "generate"
    CATEGORY = "image"
    def generate(self, width, height, batch_size=1, r=0, g=0, b=0):
        r_normalized = torch.full([batch_size, height, width, 1], r / 255.0)
        g_normalized = torch.full([batch_size, height, width, 1], g / 255.0)
        b_normalized = torch.full([batch_size, height, width, 1], b / 255.0)
        return (torch.cat((r_normalized, g_normalized, b_normalized), dim=-1),)
```

Am I doing it wrong?
